crates/router/src/compatibility/stripe/customers/types.rs

```diff
@@ -1,34 +1,86 @@
 use std::{convert::From, default::Default};
 
-use api_models::payment_methods as api_types;
+use api_models::{customers, payment_methods as api_types};
 use common_utils::{
     crypto::Encryptable,
     date_time,
     pii::{self, Email},
 };
+use masking::{Secret, SerializableSecret};
 use serde::{Deserialize, Serialize};
 
-use crate::{logger, types::api};
+use crate::{
+    logger,
+    types::{api, api::enums as api_enums},
+};
+
+impl SerializableSecret for AddressDetails {}
+
+#[derive(Default, Serialize, PartialEq, Eq, Deserialize, Clone)]
+pub struct Shipping {
+    pub address: AddressDetails,
+    pub name: Option<Secret<String>>,
+    pub carrier: Option<String>,
+    pub phone: Option<Secret<String>>,
+    pub tracking_number: Option<Secret<String>>,
+}
+
+#[derive(Default, Serialize, PartialEq, Eq, Deserialize, Clone)]
+pub struct AddressDetails {
```

Comment:
can we have a different name for this? It seems to be conflicting with others and it's not enough to give context about why it was created